hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/ObjectStore.java

```diff
@@ -299,11 +299,6 @@ public URI getKeyProviderUri() throws IOException {
 
     @Override
     public boolean hasNext() {
-      if(!currentIterator.hasNext()) {
-        currentIterator = getNextListOfVolumes(
-            currentValue != null ? currentValue.getName() : null)
-            .iterator();
-      }
```

Comment:
Thanks @nandakumar131 for this important info.

I deleted the block cause it would lead infinitely loop when I test ```listVolumes``` on my cluster.
I'm going to fix it. (without deleting it.)

Reply:
Fixed with checking existed volume.